pkg/services/accesscontrol/middleware/middleware.go

```diff
@@ -0,0 +1,45 @@
+package middleware
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"text/template"
+
+	macaron "gopkg.in/macaron.v1"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+)
+
+func Middleware(ac accesscontrol.AccessControl) func(string, ...string) macaron.Handler {
+	return func(permission string, scopes ...string) macaron.Handler {
+		return func(c *models.ReqContext) {
+			for i, scope := range scopes {
+				var buf bytes.Buffer
+
+				tmpl, err := template.New("scope").Parse(scope)
+				if err != nil {
+					c.JsonApiErr(http.StatusInternalServerError, "Internal server error", err)
```

Comment:
You should return here?
```suggestion
					c.JsonApiErr(http.StatusInternalServerError, "Internal server error", err)
					return
```